### PR TITLE
Fix bug show create view with comment

### LINKED ISSUE #4163: View does not show comment
### Steps to reproduce the behavior (Required)
1. CREATE TABLE site_access (
    datekey DATE,
    site_id INT,
    city_code SMALLINT,
    user_name VARCHAR(32),
    pv BIGINT DEFAULT '0'
)
ENGINE=olap
DUPLICATE KEY(datekey, site_id, city_code, user_name)
PARTITION BY RANGE (datekey) (
    START ("2021-01-01") END ("2021-01-04") EVERY (INTERVAL 1 day)
)
DISTRIBUTED BY HASH(site_id) BUCKETS 10
PROPERTIES (
    "replication_num" = "1" 
);
2. create view site_view (datekey COMMENT "aa", site_id, city_code) COMMENT "view comment" as select datekey, site_id, city_code from site_access;
3. show create view site_view;

### Expected behavior (Required)
 CREATE VIEW `site_view`
(
datekey COMMENT "aa",
site_id,
city_code
)
COMMENT "view comment"
AS
SELECT `default_cluster:test`.`site_access`.`datekey` AS `datekey`, `default_cluster:test`.`site_access`.`site_id` AS `site_id`, `default_cluster:test`.`site_access`.`city_code` AS `city_code` FROM `default_cluster:test`.`site_access`; 

show sql like this and it could rebuild this view.
### Real behavior (Required)
Error
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4163

## Problem Summary(Required) ：

Fix some antrl syntax and some printing on output.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 18 (11.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|0|15|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|3|66.67%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 1]
```diff
@@ -4497,8 +4497,24 @@ public static void getDdlStmt(String dbName, Table table, List<String> createTab
         // 1.1 view
         if (table.getType() == TableType.VIEW) {
             View view = (View) table;
-            sb.append("CREATE VIEW `").append(table.getName()).append("` AS ").append(view.getInlineViewDef());
-            sb.append(";");
+            sb.append("CREATE VIEW `").append(table.getName()).append("`\n(\n");
+            int idx = 0;
+            int maxIdx = table.getBaseSchema().size();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4502]: use Joiner.on(", ").join(list), code can be simplified

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 2]
```diff
@@ -4497,8 +4497,24 @@ public static void getDdlStmt(String dbName, Table table, List<String> createTab
         // 1.1 view
         if (table.getType() == TableType.VIEW) {
             View view = (View) table;
-            sb.append("CREATE VIEW `").append(table.getName()).append("` AS ").append(view.getInlineViewDef());
-            sb.append(";");
+            sb.append("CREATE VIEW `").append(table.getName()).append("`\n(\n");
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4500]: In the show statement, it is all explicit on one line, and it is not recommended to add a “\n"

**Comment by HangyuanLiu** [comment]
Please add UT to overwrite your changes

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -392,7 +392,7 @@ public ParseNode visitCreateView(StarRocksParser.CreateViewContext context) {
                 context.IF() != null,
                 targetTableName,
                 colWithComments,
-                context.viewComment == null ? null : visit(context.viewComment).toString(),
+                context.comment() == null ? null : ((StringLiteral) visit(context.comment())).getStringValue(),
```
[the remark this answers is not in the document]
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:395]: no, this will return comment string exclude keyword "COMMENT"

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|14|14|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 17 / 17 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|14|14|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 4]
```diff
@@ -4497,8 +4497,22 @@ public static void getDdlStmt(String dbName, Table table, List<String> createTab
...
+                    colSb.append(" COMMENT ").append("\"").append(column.getComment()).append("\"");
+                }
+                colDef.add(colSb.toString());
+            }
+            sb.append(Joiner.on(", ").join(colDef));
+            sb.append(")");
+            if (!Strings.isNullOrEmpty(view.getComment())) {
+                sb.append(" COMMENT \"").append(view.getComment()).append("\"");
+            }
+            sb.append(" AS ").append(view.getInlineViewDef()).append(";");
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4515]: Is the result in all one line? Is it necessary to add '\n' to be more readable?

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    private static ConnectContext connectContext;
+    private static StarRocksAssert starRocksAssert;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        FeConstants.runningUnitTest = true;
+        FeConstants.default_scheduler_interval_millisecond = 100;
+        Config.dynamic_partition_enable = true;
+        Config.dynamic_partition_check_interval_seconds = 1;
```
Comment (ABingHuang) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java:33]: Is it necessary for this ut?

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java (hunk 4 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4515]: The logic before this code is not to add line breaks, so keep this logic first, and add it later if the user needs it.

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java (hunk 5 above)
Comment (Astralidea) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowCreateViewStmtTest.java:33]: it is not necessary, these code could be remove.

**Review by wyb — APPROVED** [review]

**Pull request merged**